src/cloudWatchLogs/cloudWatchLogsUtils.ts

```diff
@@ -6,28 +6,57 @@
 import * as vscode from 'vscode'
 import { CLOUDWATCH_LOGS_SCHEME } from '../shared/constants'
 import { fromExtensionManifest } from '../shared/settings'
+import { CloudWatchLogsGroupInfo, parametersStringValue } from './registry/logStreamRegistry'
+import { CloudWatchLogsParameters } from './registry/logStreamRegistry'
 
 // URIs are the only vehicle for delivering information to a TextDocumentContentProvider.
 // The following functions are used to structure and destructure relevant information to/from a URI.
 // Colons are not valid characters in either the group name or stream name and will be used as separators.
 
+let parsingURIMap = new Map<string, URIParsingFunction>()
+
+parsingURIMap.set('viewLogStream', function (parts) {
+    return {
+        logGroupInfo: {
+            groupName: parts[1],
+            regionName: parts[2],
+            streamName: parts[3],
+        },
+        parameters: {},
+    }
+})
+
+parsingURIMap.set('searchLogGroup', function (parts) {
+    return {
+        logGroupInfo: {
+            groupName: parts[1],
+            regionName: parts[2],
+        },
+        parameters: {
+            filterPattern: parts[3],
+            startTime: Number(parts[4]),
+        },
+    }
+})
+
 /**
  * Destructures an awsCloudWatchLogs URI into its component pieces.
  * @param uri URI for a Cloudwatch Logs file
  */
-export function parseCloudWatchLogsUri(uri: vscode.Uri): { groupName: string; streamName: string; regionName: string } {
+export function parseCloudWatchLogsUri(uri: vscode.Uri): {
+    logGroupInfo: CloudWatchLogsGroupInfo
+    parameters: CloudWatchLogsParameters
+} {
     const parts = uri.path.split(':')
+    const action = parts[0]
+
+    const parsingFunction = parsingURIMap.get(action)
```

Comment:
does the sdk arn parser module help here? https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/modules/_aws_sdk_util_arn_parser.html

Reply:
It looks like this works with parsing `ARN` objects. Based on the docs, it looks like `ARN` interface includes the properties: `accountID`, `partition`, `region`, `resource`, and `service`. I am envisioning a potential solution using this where I could try to spoof the object and convert the `CloudWatchLogsData` into a `ARN` object with the properties lining up as close as possible, but this feels a little janky and could quickly become confusing to work with. It would have the advantage of coming with a parser, but it would require converting to and from this `spoofed` object which sounds like about the same amount of work as just parsing it ourselves.